bundles/org.eclipse.core.resources/src/org/eclipse/core/resources/team/ResourceRuleFactory.java

```diff
@@ -30,14 +30,24 @@
  * @since 3.0
  */
 public class ResourceRuleFactory implements IResourceRuleFactory {
-	private final IWorkspace workspace = ResourcesPlugin.getWorkspace();
+	private final IWorkspace workspace;
 
 	/**
 	 * Creates a new default resource rule factory. This constructor must only
 	 * be called by subclasses.
 	 */
 	protected ResourceRuleFactory() {
```

Comment:
Shouldn't we deprecate?

Reply:
I think this constructor should not be used at best but deprecation does incur to much discussions, complains and concerns so I don't really would like to open this can of worms here ;-)

Reply:
I don't think this one is a can of worm as it's not much used in Platform, so deprecating wouldn't add any new warning. I only see `TeamHook` that uses this constructor, but I believe that calling `ResourcesPlugin.getWorkspace().getRuleFactory()` would be better here anyway. So if you can remove this last reference to the legacy constructor and document the alternative, it can be deprecated. As long as it's not marked `forRemoval`, the deprecated method is documented about what alternative to use, and there is no reference to it in workspace, deprecating is usually welcome without discussion.
It would be a pity to miss one of the rare opportunities to quietly deprecate something in Platform ;)

Reply:
Done, this consequently means a new constructor to the `TeamHook` as well ...

Reply:
Great thanks! Only `@Deprecated` is missing.

Reply:
Done.. it seems Eclipse SDK is missing the save action here to add that automatically for lazy people like me :-D